proxy_presentation.php

```diff
@@ -67,16 +67,21 @@
         $bbbinstanceid = substr($bbbinstanceid, 0, strpos($bbbinstanceid, '['));
     }
 
-
     // Check if the user has appropriate access to the meeting resources.
-    $meetingsinstances = $DB->get_records('bigbluebuttonbn', ['meetingid' => $basemeetingid], '', 'id, course');
     $capable = false;
-    foreach ($meetingsinstances as $bbbinstance) {
-        // Checks the BBB instances to see if the user has capabilities for viewing - breaking on first success.
-        if (bigbluebuttonbn_has_capability($bbbinstance->id, 'mod/bigbluebuttonbn:view')) {
-            $capable = true;
-            break;
+    if (!bigbluebuttonbn_has_capability($bbbinstanceid, 'mod/bigbluebuttonbn:view')) {
```

Comment:
I think my approach here is correct. First of all, check the reported bbbinstanceid from the server. That will be correct 99% of the time, for activities created in place. If the perms check out, we are good to move on. Otherwise, we need to fall through and start querying for the records that match the meetingid, and then for a list of those, check the visibility of the course, then if the perms match and continue cycling until one is found.

This shortcuts the cycle in the majority of cases that work, and tightens the cases that dont.

Reply:
Ah right, I just remembered the original condition I added. If that's the case, I think it would be beneficial to also check for the $bbbinstance->id being the same as the original one passed - inside the for loop - such that it skips checking that one since it should have already been checked. The cost of doing that per should outweigh the extra capability check cost potentially.

Reply:
Updated, nice thought